### ISSUE TITLE
Extend the authorization to show some pages to public and some to users

### ISSUE THREAD

**Issue by WillGoldstein** [saleem-hadad/larecipe#158]
**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
Currently, you can either have Larecipe use Laravel's built in `auth` middleware, or not. If you elect to use it, you can then take advantage of the awesome [Gates ](https://larecipe.binarytorch.com.my/docs/2.0/authorization). But, you can't have it both ways. That is, you can't have some pages (or versions) of the docs that are open to the public, and some that are hidden for certain (or all) users.

Reasons why this would be great

1) the docs can help with SEO and be a nice way of eloquently describing features that may otherwise be a bit complex and hard to do in another setting.
2) there may be some aspects of your app that you would rather not have fall into the competitors lap so easily. We currently check each user before approving them, allowing us the opportunity to at least do our best to ensure the user is legitimate and not someone from a competing company. While this isn't perfect, it certainly makes it just a bit harder for the competition to get their hands on what we offer. Being able to hide those sensitive aspects of the docs would achieve this.

**Comment by WillGoldstein** [comment]
Just as an FYI, I got around this by taking [this](https://github.com/saleem-hadad/larecipe/issues/97#issuecomment-470075299) a step further and copying the `docs.blade.php` file and building a series of @if's within the blade to get the desired effect. It works but it's maybe not as graceful as it could be.

**Comment by WillGoldstein** [comment]
@saleem-hadad Actually, what I did here doesn't work at all.

I propose in the `__construct` function of:

-  `\vendor\binarytorch\larecipe\src\Http\Controllers\DocumentationController.php`
-  `\vendor\binarytorch\larecipe\src\Http\Controllers\SearchController.php`

change:

```
        $this->documentationRepository = $documentationRepository;

        if (config('larecipe.settings.auth')) {
            $this->middleware(['auth']);
        }
```

to

```
        $this->documentationRepository = $documentationRepository;

        if (config('larecipe.settings.auth')) {
            $this->middleware(config('larecipe.settings.middleware'));
        }
```

and in the `config/larecipe.php` file, we append `settings` so that:
```
    'settings'       => [
        'auth'       => true,
        'ga_id'      => '...',
        'middleware' => [
             'auth',
         ]
```
This way it defaults with the middleware `auth`, but lets us change it to anything we want. This is the only way I can see to have docs that can be simultaneously partially public facing, partially user facing, and partially user admin (or whatever else) facing.

**Comment by WillGoldstein** [comment]
@saleem-hadad I tried to do a pull request but couldn't make a branch since I don't have permissions I suppose. Attached are the three files I changed (it's very minor).
[proposedfix.zip](https://github.com/saleem-hadad/larecipe/files/3623548/proposedfix.zip)

**Comment by WillGoldstein** [comment]
Sorry @saleem-hadad I think this can be disregarded - using gates as described [here](https://laravel-news.com/laravel-5-7-guest-user-gates-policies) where you can have `?User $user` passed in, seems to work fine.

**Comment by WillGoldstein** [comment]
Can be closed now thanks to merging #163
